src/riscv/lib/src/machine_state/page_cache/jitted.rs

```diff
@@ -216,33 +217,35 @@ mod tests {
         assert_eq!(page.entries[100 >> 1].dispatch.called_times(), 1);
     }
 
-    #[test]
-    fn test_interpreted_fallback_on_insufficient_steps() {
-        let Ok(page) = PageEntry::<Jitted<_, M4K>, InlineCompiler>::new::<std::convert::Infallible>(
+    proptest::proptest! {
+        #[test]
+        fn test_interpreted_fallback_on_insufficient_steps(
+            max_steps in 0usize..(MAX_INSTR_COMPILED * 2)
+        ) {
+            let Ok(page) = PageEntry::<Jitted<_, M4K>, InlineCompiler>::new::<std::convert::Infallible>(
             InlineCompiler::default(),
-            |_| Ok(Instruction::new_nop(InstrWidth::Compressed)),
-        );
+                |_| Ok(Instruction::new_nop(InstrWidth::Compressed)),
+            );
 
         let mut core = MachineCoreState::new();
```

Comment:
```suggestion
            let Ok(page) = PageEntry::<Jitted<_, M4K>, InlineCompiler>::new::<std::convert::Infallible>(
                InlineCompiler::default(),
                |_| Ok(Instruction::new_nop(InstrWidth::Compressed)),
            );

            let mut core = MachineCoreState::new();
```

Our linting doesn't work inside the `proptest!` macro so we have to do it manually. 🤦‍♂️ I think this is following our usual practice?